src/vcpkg/base/files.cpp

```diff
@@ -517,6 +517,19 @@ namespace vcpkg::Files
                 linfo, "error renaming file: %s: %s: %s", fs::u8string(oldpath), fs::u8string(newpath), ec.message());
         }
     }
+    void Filesystem::rename_with_retry(const fs::path& oldpath, const fs::path& newpath, std::error_code& ec)
+    {
+        using namespace std::chrono_literals;
+        std::chrono::milliseconds delays[] = {0ms, 100ms, 1000ms};
+        size_t max_retry = sizeof(delays) / sizeof(std::chrono::milliseconds);
+        size_t retry = 0;
+        do
+        {
+            std::this_thread::sleep_for(delays[retry]);
+            this->rename(oldpath, newpath, ec);
+            ++retry;
+        } while (ec && retry < max_retry);
```

Comment:
```suggestion
        this->rename(oldpath, newpath, ec);
        using namespace std::chrono_literals;
        std::chrono::milliseconds delays[] = {10ms, 100ms, 1000ms};
        for (const auto& delay : delays) {
            if (!ec) {
                return;
            }

            std::this_thread::sleep_for(delay);
            this->rename(oldpath, newpath, ec);
        }
```

I think it would be good to avoid calling sleep_for for the first attempt with a 0.

Reply:
> ```for (const auto& delay : {10ms, 100ms, 1000ms})```
oh, I really ❤️  this way of writing the loop! We don't even have to compute the number of elements.
Good suggestion.